lib/std/heap/PageAllocator.zig

```diff
@@ -39,8 +40,8 @@ fn alloc(_: *anyopaque, n: usize, log2_align: u8, ra: usize) ?[*]u8 {
         -1,
         0,
     ) catch return null;
-    assert(mem.isAligned(@intFromPtr(slice.ptr), mem.page_size));
-    const new_hint: [*]align(mem.page_size) u8 = @alignCast(slice.ptr + aligned_len);
+    assert(mem.isAligned(@intFromPtr(slice.ptr), heap.pageSize()));
+    const new_hint: [*]u8 = mem.alignPointer(slice.ptr + aligned_len, heap.pageSize()) orelse return null;
```

Comment:
This `return null` leaks memory and is highly sus

Reply:
So an error is better?

Reply:
Ok, it unmaps on failure.